test/tools/ossfuzz/protoToSol.h

```diff
@@ -0,0 +1,185 @@
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#include <test/tools/ossfuzz/solProto.pb.h>
+
+#include <random>
+#include <string>
+#include <utility>
+#include <variant>
+
+namespace solidity::test::solprotofuzzer
+{
+/// Random number generator that is seeded with a fuzzer
+/// supplied unsigned integer.
+struct SolRandomNumGenerator
+{
+	using RandomEngine = std::minstd_rand;
+
+	explicit SolRandomNumGenerator(unsigned _seed): m_random(RandomEngine(_seed)) {}
+
+	/// @returns a pseudo random unsigned integer
+	unsigned operator()()
+	{
+		return m_random();
+	}
+
+	RandomEngine m_random;
+};
+
+/* There are two types of tests created by the converter:
+ * - library test
+ * - contract test
+ *
+ * The template for library test is the following:
+ *
+ * // Library generated from fuzzer protobuf specification
+ * library L0 {
+ *      function f0(uint) public view returns (uint) {
+ *          return 31337;
+ *      }
+ *      function f1(uint) public pure returns (uint) {
+ *          return 455;
+ *      }
+ * }
+ * library L1 {
+ *      function f0(uint) external view returns (uint) {
+ *          return 607;
+ *      }
+ * }
+ *
+ * // Test entry point
+ * contract C {
+ *     // Uses a single pseudo randomly chosen library
+ *     // and calls a pseudo randomly chosen function
+ *     // returning a non-zero error code on failure or
+ *     // a zero uint when test passes.
+ *     using L0 for uint;
+ *     function test() public pure returns (uint) {
+ *          uint x;
+ *          if (x.f1() != 455)
+ *              return 1;
+ *          return 0;
+ *     }
+ * }
+ *
+ * The template for contract test is the following
+ * // Contracts generated from fuzzer protobuf specification
+ * contract C0B {
+ *      function f0() public pure virtual returns (uint)
+ *      {
+ *          return 42;
+ *      }
+ * }
+ * contract C0 is C0B {
+ *      function f0() public pure override returns (uint)
+ *      {
+ *          return 1337;
+ *      }
+ * }
+ *
+ * // Test entry point
+ * contract C {
+ *      // Invokes one or more contract functions returning
+ *      // a non-zero error code for failure, a zero uint
+ *      // when all tests pass
+ *      function test() public pure returns (uint)
+ *      {
+ *          C0 tc0 = new C0();
+ *          if (tc0.f0() != 1337)
+ *              return 1;
+ *          C0B tc1 = new C0B();
+ *          if (tc1.f0() != 42)
+ *              return 2;
+ *          // Expected return value if all tests pass
+ *          return 0;
+ *      }
+ * }
+ */
+class ProtoConverter
+{
+public:
+	ProtoConverter() {}
+	ProtoConverter(ProtoConverter const&) = delete;
+	ProtoConverter(ProtoConverter&&) = delete;
```

Comment:
Why delete these 2?

Reply:
Because there is no use case for copying/moving a converter class.

Reply:
Right, but would you explicitly disallow that? Or is it just a "doesn't matter"?

Reply:
Initially, I preferred doesn't matter. But at some point @chriseth suggested to make this explicit I think. After that, I've essentially tried to keep it consistent.